### PR TITLE
Update steps on how to make a new API version

### PR DESCRIPTION
**Pull request by msau42**
Also add guidance on how to structure commits when doing API promotions.

### CODE REVIEW

File: contributors/devel/sig-architecture/api_changes.md [hunk 1]
```diff
@@ -658,23 +658,35 @@ If you are adding a new API version to an existing group, you can copy the
...
-* You must add the new version  to
-  [hack/update-generated-protobuf-dockerized.sh](https://github.com/kubernetes/kubernetes/blob/v1.8.2/hack/update-generated-protobuf-dockerized.sh#L44)
-  to generate protobuf IDL and marshallers.
+  [hack/lib/init.sh#KUBE_AVAILABLE_GROUP_VERSIONS](https://github.com/kubernetes/kubernetes/blob/v1.21.2/hack/lib/init.sh#L65).
 * You must add the new version  to
-  [cmd/kube-apiserver/app#apiVersionPriorities](https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/cmd/kube-apiserver/app/aggregator.go#L172)
+  [cmd/kube-apiserver/app#apiVersionPriorities](https://github.com/kubernetes/kubernetes/blob/v1.21.2/cmd/kube-apiserver/app/aggregator.go#L247).
 * You must setup storage for the new version in
-  [pkg/registry/group_name/rest](https://github.com/kubernetes/kubernetes/blob/v1.8.0-alpha.2/pkg/registry/authentication/rest/storage_authentication.go)
+  [pkg/registry/group_name/rest](https://github.com/kubernetes/kubernetes/blob/v1.21.2/pkg/registry/authentication/rest/storage_authentication.go).
```
Comment (liggitt) [inline contributors/devel/sig-architecture/api_changes.md:681]:
```suggestion
pkg/registry/<group>/rest (for example, [pkg/registry/authentication/rest/storage_authentication.go](https://github.com/kubernetes/kubernetes/blob/v1.21.2/pkg/registry/authentication/rest/storage_authentication.go).
```

File: contributors/devel/sig-architecture/api_changes.md [hunk 2]
```diff
@@ -658,23 +658,35 @@ If you are adding a new API version to an existing group, you can copy the
...
 
 You need to regenerate the generated code as instructed in the sections above.
 
+Afterwards, sanity test the changes by bringing up a cluster (i.e.,
+local-up-cluster.sh, kind, etc) and running `kubectl get
+<resource>.<version>.<group>`.  [Integration tests](../sig-testing/integration-tests.md)
+are also good for testing the full CRUD lifecycle along with the controller.
+
```
Comment (liggitt) [inline contributors/devel/sig-architecture/api_changes.md:689]: probably worth noting two specific tests that are known to require data updates:

```
storage version hashes published in API discovery data
* update data in `pkg/controlplane/storageversionhashdata/data.go#GVRToStorageVersionHash`
* run `go test ./pkg/controlplane -run StorageVersion` to verify 

persisted versions stored in etcd
* add a version stub for the new versions in `test/integration/etcd/data.go`
* run `go test ./test/integration/etcd` to verify
```

File: contributors/devel/sig-architecture/api_changes.md [hunk 3]
```diff
@@ -658,23 +658,35 @@ If you are adding a new API version to an existing group, you can copy the
 structure of the existing `pkg/apis/<group>/<existing-version>` and
 `staging/src/k8s.io/api/<group>/<existing-version>` directories.
 
+It is helpful to structure the PR in layered commits to make it easier for
+reviewers to see what has changed between the two versions:
+1. A commit that just copies the `pkg/apis/<group>/<existing-version>` and
+   `staging/src/k8s.io/api/<group>/<existing-version>` packages to the
+   `<new-version>`.
+1. A commit that renames `<existing-version>`to `<new-version>` in the new files.
+1. A commit that makes any new changes for `<new-version>`.
```
Comment (liggitt) [inline contributors/devel/sig-architecture/api_changes.md:667]: and then a commit containing generated changes (output from `make generated_files`, `make update` etc)

**Comment by liggitt** [comment]
thanks for opening this, just a couple suggestions, then lgtm

**Comment by msau42** [comment]
Comments addressed

**Comment by liggitt** [comment]
thanks!

/lgtm
/approve

**Pull request merged**